### PR TITLE
[Graphbolt] Change data_block to mini_batch

### PR DESCRIPTION
**Pull request by peizhou001**
## Description
Change `data_block` to `minibatch`, and merge link and node block.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/subgraph_sampler.py [hunk 1]
```diff
@@ -28,24 +27,30 @@ def __init__(
...
-                data.compacted_node_pair,
-                data.compacted_negative_head,
-                data.compacted_negative_tail,
-            ) = self._link_prediction_preprocess(data)
-        elif isinstance(data, NodeClassificationBlock):
-            seeds = data.seed_node
+                minibatch.compacted_node_pair,
+                minibatch.compacted_negative_head,
+                minibatch.compacted_negative_tail,
+            ) = self._link_prediction_preprocess(minibatch)
```
Comment (frozenbugs) [inline python/dgl/graphbolt/subgraph_sampler.py:37]:
```suggestion
            ) = self._node_pair_preprocess(minibatch)
```

File: python/dgl/graphbolt/minibatch.py [hunk 2]
```diff
@@ -69,17 +63,12 @@ class NodeClassificationBlock(DataBlock):
     label: Union[torch.Tensor, Dict[str, torch.Tensor]] = None
```
Comment (frozenbugs) [inline python/dgl/graphbolt/minibatch.py:63]: duplicated label in L85, remove one.

File: tests/python/pytorch/graphbolt/test_subgraph_sampler.py [hunk 3]
```diff
@@ -6,26 +6,21 @@
...
-
 @pytest.mark.parametrize("labor", [False, True])
 def test_SubgraphSampler_Node(labor):
     graph = gb_test_utils.rand_csc_graph(20, 0.15)
     itemset = gb.ItemSet(torch.arange(10))
     minibatch_dp = gb.MinibatchSampler(itemset, batch_size=2)
     num_layer = 2
     fanouts = [torch.LongTensor([2]) for _ in range(num_layer)]
-    data_block_converter = Mapper(minibatch_dp, to_node_block)
+    data_block_converter = Mapper(minibatch_dp, gb_test_utils.to_node_minibatch)
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/test_subgraph_sampler.py:16]: minibatch_collator

File: tests/python/pytorch/graphbolt/test_subgraph_sampler.py [hunk 4]
```diff
@@ -41,7 +36,7 @@ def test_SubgraphSampler_Link(labor):
     minibatch_dp = gb.MinibatchSampler(itemset, batch_size=2)
     num_layer = 2
     fanouts = [torch.LongTensor([2]) for _ in range(num_layer)]
-    data_block_converter = Mapper(minibatch_dp, to_link_block)
+    data_block_converter = Mapper(minibatch_dp, gb_test_utils.to_link_minibatch)
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/test_subgraph_sampler.py:39]: ditto and others.

File: python/dgl/graphbolt/impl/neighbor_sampler.py [hunk 5]
```diff
@@ -54,7 +54,7 @@ def __init__(
         >>> import dgl.graphbolt as gb
         >>> from torchdata.datapipes.iter import Mapper
         >>> def to_link_block(data):
-            ... block = gb.LinkPredictionBlock(node_pair=data)
+            ... block = gb.MiniBatch(node_pair=data)
```
Comment (frozenbugs) [inline python/dgl/graphbolt/impl/neighbor_sampler.py:57]:
```suggestion
            ... minibatch = gb.MiniBatch(node_pair=data)
```

File: python/dgl/graphbolt/negative_sampler.py [hunk 6]
```diff
@@ -43,18 +43,18 @@ def _sample(self, data):
...
         Returns
         -------
-        LinkPredictionBlock
-            An instance of 'LinkPredictionBlock' encompasses both positive and
-            negative samples.
+        MiniBatch
+            An instance of 'MiniBatch' encompasses both positive and negative
+            samples.
         """
         node_pairs = data.node_pair
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/negative_sampler.py:59]: `assert node_pairs is not None`?

File: tests/python/pytorch/graphbolt/impl/test_negative_sampler.py [hunk 7]
```diff
@@ -22,7 +18,9 @@ def test_NegativeSampler_Independent_Format(negative_ratio):
     )
     batch_size = 10
     item_sampler = gb.ItemSampler(item_set, batch_size=batch_size)
-    data_block_converter = Mapper(item_sampler, to_data_block)
+    data_block_converter = Mapper(
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/impl/test_negative_sampler.py:21]: rename `data_block_converter`?

File: tests/python/pytorch/graphbolt/impl/test_negative_sampler.py [hunk 8]
```diff
@@ -55,7 +53,9 @@ def test_NegativeSampler_Conditioned_Format(negative_ratio):
...
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/impl/test_negative_sampler.py:56]: rename

File: tests/python/pytorch/graphbolt/test_feature_fetcher.py [hunk 9]
```diff
@@ -19,7 +19,9 @@ def test_FeatureFetcher_homo():
     item_sampler_dp = gb.ItemSampler(itemset, batch_size=2)
     num_layer = 2
     fanouts = [torch.LongTensor([2]) for _ in range(num_layer)]
-    data_block_converter = Mapper(item_sampler_dp, gb_test_utils.to_node_block)
+    data_block_converter = Mapper(
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/test_feature_fetcher.py:22]: rename

File: tests/python/pytorch/graphbolt/test_multi_process_dataloader.py [hunk 10]
```diff
@@ -23,7 +22,9 @@ def test_DataLoader():
     feature_store = dgl.graphbolt.BasicFeatureStore(features)
 
     item_sampler = dgl.graphbolt.ItemSampler(itemset, batch_size=B)
-    block_converter = Mapper(item_sampler, gb_test_utils.to_node_block)
+    block_converter = Mapper(
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/test_multi_process_dataloader.py:25]: rename

File: tests/python/pytorch/graphbolt/test_single_process_dataloader.py [hunk 11]
```diff
@@ -25,7 +20,9 @@ def test_DataLoader():
     feature_store = dgl.graphbolt.BasicFeatureStore(features)
 
     item_sampler = dgl.graphbolt.ItemSampler(itemset, batch_size=B)
-    block_converter = Mapper(item_sampler, to_node_block)
+    block_converter = Mapper(
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/test_single_process_dataloader.py:23]: rename

File: tests/python/pytorch/graphbolt/test_subgraph_sampler.py [hunk 12]
```diff
@@ -6,26 +6,23 @@
...
-
 @pytest.mark.parametrize("labor", [False, True])
 def test_SubgraphSampler_Node(labor):
     graph = gb_test_utils.rand_csc_graph(20, 0.15)
     itemset = gb.ItemSet(torch.arange(10))
     item_sampler_dp = gb.ItemSampler(itemset, batch_size=2)
     num_layer = 2
     fanouts = [torch.LongTensor([2]) for _ in range(num_layer)]
-    data_block_converter = Mapper(item_sampler_dp, to_node_block)
+    data_block_converter = Mapper(
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/test_subgraph_sampler.py:16]: rename

**Review by Rhett-Ying — APPROVED** [review]
LGTM except for names like `data_block_xx`. Please remove any occurrences of `data_block` from graphbolt.

File: python/dgl/graphbolt/negative_sampler.py [hunk 13]
```diff
@@ -99,7 +100,7 @@ def _collate(self, data, neg_pairs, etype=None):
 
         Parameters
         ----------
-        data : LinkPredictionBlock
+        data : MiniBatch
```
Comment (frozenbugs) [inline python/dgl/graphbolt/negative_sampler.py:103]: data -> minibatch?

**Review by frozenbugs — APPROVED** [review]
